src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIteratorExtensions.cs

```diff
@@ -150,7 +150,8 @@ public static string GetUtf8String(this MemoryPoolIterator start, MemoryPoolIter
             var decoder = _utf8.GetDecoder();
 
             var length = start.GetLength(end);
-            var charLength = length * 2;
+            var charLength = length;
```

Comment:
@davidfowl you need to brush up on your C/C++; that pattern is everywarez! But they also have `long long`s and an `int` of undefined size :disappointed: